source/backends/ubuntu/ubupkg.d

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ * Author: Iain Lane <iain.lane@canonical.com>
+ *
+ * Licensed under the GNU Lesser General Public License Version 3
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the license, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module backends.ubuntu.ubupkg;
+
+import appstream.Component;
+
+import backends.debian.debpkg;
+import bindings.gettext;
+import backends.interfaces;
+
+import glib.Internationalization;
+import glib.KeyFile;
+
+import std.container : Array;
+
+import logging;
+
+immutable DESKTOP_GROUP = "Desktop Entry";
+
+class UbuntuPackage : DebPackage
+{
+    this (string pname, string pver, string parch, string globalTmpDir, ref Array!Package allPackages)
+    {
+        this.globalTmpDir = globalTmpDir;
+        this.langpackDir = buildPath (globalTmpDir, "langpacks");
```

Comment:
How big are the langpacks? Does it make sense to unpack them into an in-memory location for speed?